pkg/capabilities/consensus/ocr3/reporting_plugin.go

```diff
@@ -216,6 +218,16 @@ func (r *reportingPlugin) Outcome(outctx ocr3types.OutcomeContext, query types.Q
 		o.Outcomes[weid.WorkflowId] = outcome
 	}
 
+	// Prune previous outcomes for deleted workflows
+	var prunedWorkflows []string
+	for _, wfID := range r.r.getDeregisteredWorkflows() {
```

Comment:
I'm afraid this is not going to be that simple. Outcome() method needs to be pure, not related to any local state. We need to receive 2F+1 identical outcomes, otherwise we risk deadlocking OCR. 

One idea I had is to ask nodes to send a full list of workflow IDs that they think are still alive inside their observations. Then in the Outcome we could weed out the ones that have fewer than F+1 votes. @cedric-cordenier thoughts?

Reply:
Understood

Reply:
@bolekk 🤦  So we're effectively getting consensus on the workflows we want to prune. That makes sense I think, but doesn't that mean we would effectively be propagating the deregistration? Is that desirable? I think that could have weird side effects for running executions, but maybe that's OK

Reply:
Execution would fail the consensus step anyway, if enough nodes already deleted that workflow. I'm open to other approaches, I just don't see any good alternatives atm.